src/main/java/controller/TimetableController.java

```diff
@@ -54,13 +56,19 @@ public void findTimetableForBusNumber(int busNumber) {
         Statement statement = databaseHandler.createStatement();
         try {
             String findTimetable = "\tSELECT * FROM timetable\n" +
-                    "\tINNER JOIN buses on buses.driverId = drivers.id\n" +
+                    "\tINNER JOIN buses on buses.id = timetable.busId\n" +
                     "\tWHERE buses.busNumber = " + busNumber;
             ResultSet busTimetable = statement.executeQuery(findTimetable);
+            while (busTimetable.next()){
```

Comment:
You do not need a while here, only one if statement, because resultSet anyway gives you one result.
this method I would rewrite like this:
```
public void findTimetableForBusNumber(String busNumber) {
        Statement statement = databaseHandler.createStatement();
        try {
            String findTimetable = "\tSELECT * FROM timetable\n" +
                    "\tINNER JOIN buses on buses.id = timetable.busId\n" +
                    "\tWHERE buses.busNumber = '" + busNumber + "' limit 1";
            ResultSet busTimetable = statement.executeQuery(findTimetable);
            if (busTimetable.next()) {
                String myBusWeekDay = busTimetable.getString("weekday");
                String myBusArrivalTime = busTimetable.getString("arrivalTime");
                int myBusLocationId = busTimetable.getInt("locationId");
                System.out.println("Here is timetable for bus number " + busNumber
                        + ": weekday - " + myBusWeekDay + ", arrival time - " + myBusArrivalTime
                        + " location - " + myBusLocationId);
            } else
                    System.out.println("Bus or timetable does not exist");

            statement.close();
            databaseHandler.closeConnection();

        }catch (SQLException e) {
            System.out.println(e);
        }
    }
```

Reply:
agreed, had some kind of error and that is how I resolved it :D 
but now I see your code works just fine, will update it.